src/builder/actions/Actions.sol

```diff
@@ -676,6 +687,100 @@ library Actions {
         return (quarkOperation, action);
     }
 
+    function bridgeAcross(BridgeAsset memory bridge, PaymentInfo.Payment memory payment, bool useQuotecall)
+        internal
+        view
+        returns (IQuarkWallet.QuarkOperation memory, Action memory)
+    {
+        Accounts.ChainAccounts memory srcChainAccounts =
+            Accounts.findChainAccounts(bridge.srcChainId, bridge.chainAccountsList);
+
+        Accounts.ChainAccounts memory dstChainAccounts =
+            Accounts.findChainAccounts(bridge.destinationChainId, bridge.chainAccountsList);
+
+        Accounts.AssetPositions memory srcAssetPositions =
+            Accounts.findAssetPositions(bridge.assetSymbol, srcChainAccounts.assetPositionsList);
+
+        Accounts.AssetPositions memory dstAssetPositions =
+            Accounts.findAssetPositions(bridge.assetSymbol, dstChainAccounts.assetPositionsList);
+
+        Accounts.QuarkSecret memory accountSecret =
+            Accounts.findQuarkSecret(bridge.sender, srcChainAccounts.quarkSecrets);
+
+        bytes[] memory scriptSources = new bytes[](1);
+        scriptSources[0] = Across.bridgeScriptSource();
+
+        // Make FFI call to fetch a quote from Across API
+        bytes memory ffiCalldata = abi.encodeCall(
```

Comment:
In `FFI.sol`, I wonder if we should have something akin to `IAcrossFFI` as well as `AcrossFFI` and just call `AcrossFFI.requestQuote(..)` and then this logic (and the FFI contract address) moves there. It would probably be a slightly neater way to bundle this from the code-caller's perspective.

Reply:
Good suggestion! I moved this logic out into `FFI.sol`. The FFI addresses are also defined there now.